impl/src/main/java/com/okta/sdk/impl/client/DefaultClientBuilder.java

```diff
@@ -329,8 +392,55 @@ public ClientBuilder setOrgUrl(String baseUrl) {
         return this;
     }
 
+    @Override
+    public ClientBuilder setAuthorizationMode(String authorizationMode) {
+        // if 'authorizationMode' is ever supplied, it should only be "PrivateKey" (OAUTH2 flow);
+        // else do not set it as it defaults to SSWS flow.
+        if (authorizationMode != null) {
+            if (authorizationMode.equals("PrivateKey")) {
+                this.clientConfig.setAuthorizationMode(authorizationMode);
+            }
+            else {
+                throw new IllegalArgumentException("Invalid authorizationMode");
+            }
+        }
+        return this;
+    }
+
+    @Override
+    public ClientBuilder setScopes(List<String> scopes) {
+        if (isOAuth2Flow()) {
+            Assert.notNull(scopes, "Missing scopes");
+            Assert.isTrue(scopes.size() > 0, "Empty scopes list");
+            this.clientConfig.setScopes(scopes);
+        }
+        return this;
+    }
+
+    @Override
+    public ClientBuilder setPrivateKey(String privateKey) {
+        if (isOAuth2Flow()) {
```

Comment:
IMHO, calling `setPrivateKey` should be all that is needed to enable the oauth flow.

Reply:
I'm not sure about it. Personally, I like to have everything in my configuration file (API token and PrivateKey) and I can easily use the strategy I want just by setting the AuthorizationMode.

Reply:
@laura-rodriguez no objection, there but we could default one taking priority if both are defined, and `AuthorizationMode` is not set. (or fail fast if both are set)

Reply:
We agreed to use SSWS as default for `AuthorizationMode`: https://github.com/okta/oss-technical-designs/blob/master/technical_designs/oauth-in-management-sdks.md#authorizationmode-enum

Reply:
@bdemers see my comment above on why we had to use _authorizationMode_ attribute.